src/components/AddStaffMemberModal.vue

```diff
@@ -223,7 +223,13 @@ export default defineComponent({
     },
     getPartyRoleTypeId(partyId: string) {
       return this.getParty(partyId) ? this.getParty(partyId).roleTypeId : ''
-    }
+    },
+    isRoleUpdated() {
+      return this.parties.some((party: any) => {
+        const selectedParty = this.selectedPartyValues.find((p: any) => p.partyId === party.partyId);
+        return selectedParty?.roleTypeId !== party.roleTypeId;
```

Comment:
When fetching parties to be displayed in the modal we are not fetching roleTypeId, thus if we get a party in the list whose role is already selected the button will become active irrespective of whether any change is made in the modal or not.